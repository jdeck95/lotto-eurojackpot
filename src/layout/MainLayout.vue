<script setup lang="ts">
import NavBar from '@/components/layout/NavBar.vue';
</script>

<template>
  <div>
    <q-layout view="lhr lpR lfr">
      <q-header>
        <NavBar />
      </q-header>
      <q-page-container class="row justify-center">
        <router-view />
      </q-page-container>
      <q-footer class="footer q-pa-sm">Images and texts were created with AI</q-footer>
    </q-layout>
  </div>
</template>

<style scoped lang="scss">
.footer {
  background-color: #fcbdc0;
  color: black;
}
</style>
