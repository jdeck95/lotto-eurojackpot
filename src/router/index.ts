import { createRouter, createWebHistory } from 'vue-router';
import HomeView from '@/views/HomeView.vue';
import MainLayout from '@/layout/MainLayout.vue';

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    {
      path: '/',
      name: 'home',
      meta: { layout: MainLayout },
      component: HomeView,
    },
    {
      path: '/eurojackpot',
      name: 'eurojackpot',
      meta: { layout: MainLayout },
      component: () => import('../views/EurojackpotView.vue'),
    },
  ],
});

export default router;
