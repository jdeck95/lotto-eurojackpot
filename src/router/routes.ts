import HomeView from '@/views/HomeView.vue';
import MainLayout from '@/layout/MainLayout.vue';

export const routes = [
  {
    path: '/',
    name: 'home',
    label: 'Start',
    meta: { layout: MainLayout },
    component: HomeView,
  },
  {
    path: '/eurojackpot',
    name: 'eurojackpot',
    label: 'Eurojackpot',
    meta: { layout: MainLayout },
    component: () => import('../views/EurojackpotView.vue'),
  },
];
