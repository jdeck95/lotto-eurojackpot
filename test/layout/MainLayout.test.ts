import { mount, type VueWrapper } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import { setActivePinia } from 'pinia';
import { Quasar } from 'quasar';
import { beforeEach, describe, expect, it } from 'vitest';
import MainLayout from '../../src/layout/MainLayout.vue';
import router from '@/router';

describe('MainLayout', () => {
  let wrapper: VueWrapper<any>;

  const pinia = createTestingPinia();
  setActivePinia(pinia);

  beforeEach(() => {
    wrapper = mount(MainLayout, {
      global: {
        plugins: [Quasar, pinia, router],
      },
    });
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
