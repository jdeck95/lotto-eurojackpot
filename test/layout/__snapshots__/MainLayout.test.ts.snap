// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainLayout > should render correctly 1`] = `
"<div data-v-fedce037="">
  <div data-v-fedce037="" class="q-layout q-layout--standard" style="min-height: 768px;" tabindex="-1">
    <!----><object class="q--avoid-card-border" style="display: block; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;" tabindex="-1" type="text/html" data="about:blank" aria-hidden="true"></object>
    <header data-v-fedce037="" class="q-header q-layout__section--marginal absolute-top header">
      <div data-v-d34746f4="" data-v-fedce037="" class="q-toolbar row no-wrap items-center toolbar text-white justify-center" role="toolbar">
        <div data-v-d34746f4="" class="row no-wrap"><img data-v-d34746f4="" src="/src/assets/wow-logo.png" alt="Wheel Of Wealth Logo" width="140px" hight="100px">
          <div data-v-d34746f4="" class="column justify-center title q-mr-lg">Wheel Of Wealth</div>
        </div><a data-v-d34746f4="" href="/" class="q-mr-md">Start</a><a data-v-d34746f4="" href="/eurojackpot" class="q-mr-md">Eurojackpot</a>
      </div><object class="q--avoid-card-border" style="display: block; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;" tabindex="-1" type="text/html" data="about:blank" aria-hidden="true"></object>
    </header>
    <div data-v-fedce037="" class="q-page-container row justify-center" style="padding-top: 50px; padding-bottom: 50px;">
      <!---->
    </div>
    <footer data-v-fedce037="" class="q-footer q-layout__section--marginal absolute-bottom footer q-pa-sm"><object class="q--avoid-card-border" style="display: block; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;" tabindex="-1" type="text/html" data="about:blank" aria-hidden="true"></object>Images and texts were created with AI</footer>
  </div>
</div>"
`;
